script/tool/lib/src/version_check_command.dart

```diff
@@ -208,38 +210,65 @@ ${indentation}HTTP response: ${pubVersionFinderResponse.httpResponse.body}
           pubspec.name.endsWith('_platform_interface');
       if (isPlatformInterface &&
           allowedNextVersions[headVersion] == NextVersionType.BREAKING_MAJOR) {
-        final String error = '$pubspecPath breaking change detected.\n'
-            'Breaking changes to platform interfaces are strongly discouraged.\n';
-        printErrorAndExit(errorMessage: error);
+        printError('$pubspecPath breaking change detected.\n'
+            'Breaking changes to platform interfaces are strongly discouraged.\n');
+        badVersionChangePubspecs.add(pubspecPath);
+        continue;
       }
     }
+    _pubVersionFinder.httpClient.close();
 
+    // TODO(stuartmorgan): Unify the way iteration works for these checks; the
+    // two checks shouldn't be operating independently on different lists.
+    final List<String> mismatchedVersionPlugins = <String>[];
     await for (final Directory plugin in getPlugins()) {
-      await _checkVersionsMatch(plugin);
+      if (!(await _checkVersionsMatch(plugin))) {
+        mismatchedVersionPlugins.add(plugin.basename);
+      }
+    }
+
+    bool passed = true;
+    if (badVersionChangePubspecs.isNotEmpty) {
+      passed = false;
+      printError('''
+The following pubspecs failed validaton:
+$indentation${badVersionChangePubspecs.join('\n$indentation')}
+''');
+    }
+    if (mismatchedVersionPlugins.isNotEmpty) {
+      passed = false;
+      printError('''
+The following pubspecs failed validaton:
+$indentation${badVersionChangePubspecs.join('\n$indentation')}
```

Comment:
This seems need to be a different message? Something like: "The following plugins' versions on CHANGELOG and pubspec.yaml are not matched."

Reply:
I was printing the wrong list of plugins here too; thanks for catching the copypasta!